lamindb_setup/dev/_settings_instance.py

```diff
@@ -111,10 +115,32 @@ def _update_local_sqlite_file(self) -> None:
                 "updating local SQLite & locking cloud SQLite (sync back & unlock:"
                 " lamin close)"
             )
+            self._cloud_sqlite_locker.lock()
```

Comment:
This is not needed, `_check_sqlite_lock()` will call `lock()` under the hood.

Reply:
In `.has_lock`? This seems very implicit.

Reply:
I see:
<img width="426" alt="image" src="https://github.com/laminlabs/lamindb-setup/assets/16916678/1c8ff057-bd9b-400b-9e97-244c21c83052">

Reply:
yep there

Reply:
I also think it makes sense logically: One first tries to lock but as this might not work without throwing an error one calls an explicit check.

Reply:
I can read this code and easily understand it even though it's logic from you. I always think that's a good test for good code.